factory-containers/assemble-system-image

```diff
@@ -0,0 +1,151 @@
+        parser.print_help()
+        exit(1)
+
+    if args.build_num is None:
+        logger.error('Argument `Build number` is missing, specify it either as a command line argument'
+                     ' or an H_BUILD environment variable')
+        parser.print_help()
+        exit(1)
+
+    if args.token_file is None:
+        logger.error('Argument `Token file` is missing, specify it either as a command line argument'
+                     ' or put it into `/secrets/osftok`')
+        parser.print_help()
+        exit(1)
+
+    else:
+        with open(args.token_file) as token_file:
+            vars(args)['token'] = token_file.read().strip()
+
+    if args.wic_tool is None:
+        logger.error('Argument `WIC tool` is missing, specify it either as a command line argument'
+                     ' or a WIC_TOOL environment variable')
+        parser.print_help()
+        exit(1)
+
+    if args.app_image_dir is None:
+        logger.error('Argument `App Image Dir` is missing, specify it either as a command line argument'
+                     ' or a APP_IMAGE_DIR environment variable')
+        parser.print_help()
+        exit(1)
+
+    if args.out_image_dir is None:
+        logger.error('Argument `Out Image Dir` is missing, specify it either as a command line argument'
+                     ' or a OUT_IMAGE_DIR environment variable')
+        parser.print_help()
+        exit(1)
+
+    return args
+
+
+if __name__ == '__main__':
+    logging.basicConfig(level=logging.INFO)
+    args = get_args()
+
+    logger.info('Fetching factory targets; factory: {}, version: {}'.format(args.factory, args.build_num))
+    targets = get_targets_from_api(args.factory, args.build_num, args.token)
+
+    for target in targets:
+        if len(target) != 1:
+            raise Exception('Got invalid Targets from TUF server: ' + str(targets))
+
+        target_name = next(iter(target))
+        logger.info('Got Target {}, processing it...'.format(target_name))
+
+        image_base_url = target[target_name]['custom']['uri']
+        image_machine = target[target_name]['custom']['hardwareIds'][0]
+        # TODO: Fix hardcoding
+        image_name = 'lmp-factory-image'
+        wic_image = '{}-{}.wic'.format(image_name, image_machine)
+
+        logger.info('Fetching a WIC image for Target: {}'.format(target_name))
+        get_system_image_from_ci(image_base_url, image_name, image_machine, args.token)
+
+        containers_sha = target[target_name]['custom']['containers-sha']
+
+        #containers_arch = target[target_name]['custom']['arch']
+        # TODO: Fix:  proper solution for finding
+        containers_arch = 'arm64' if image_machine == 'raspberrypi3-64' else 'amd64'
+
+        app_image_tar_src = os.path.join(args.app_image_dir, containers_sha, '{}-{}.tar'.
+                                         format(containers_sha, containers_arch))
+        app_image_tar_dst = wic_image + ':2/ostree/deploy/lmp/var/sota/import/'
```

Comment:
so we wind up copying the tarfile to /var/sota/import/. i guess we then do something with that on first boot?

Reply:
yes, still considering options here, either untar it on the first boot (aklite when it imports 'installed_versions' on the first boot) or actually try to put into the wic image all untared files, the issue here is that the wic tool doesn't want to copy folder.